paf-mvp-frontend/tests/test-cases/queue.test.ts

```diff
@@ -0,0 +1,88 @@
+import { TestWatcher } from 'jest';
+import { isExportDeclaration } from 'typescript';
+import { Command, CommandQueue, processCommands } from '../../src/utils/queue';
+
+class CommandMocker {
+  public called = false;
+
+  func(): Command {
+    return () => {
+      this.called = true;
+    };
+  }
+}
+
+describe('queue', () => {
+  let cmd1: CommandMocker;
+  let cmd2: CommandMocker;
+  let cmd3: CommandMocker;
+  const lib = {
+    queue: [] as CommandQueue,
+  };
+
+  beforeEach(() => {
+    cmd1 = new CommandMocker();
+    cmd2 = new CommandMocker();
+    cmd3 = new CommandMocker();
+  });
+
+  test('processCommands undefined do not crash', () => {
+    lib.queue = processCommands(undefined);
+  });
+
+  test('processCommands empty array do nothing', () => {
+    lib.queue = processCommands(lib.queue);
+  });
+
+  test('processCommands 2 commands', () => {
+    lib.queue = processCommands([cmd1.func(), cmd2.func()]);
+
+    expect(cmd1.called).toBeTruthy();
```

Comment:
Using `jest.fn()` you can test not only that the method was called, but that it was called **once**, which, I believe, is very important.

Reply:
Indeed. A counter is better.